lib/storage/index.ts

```diff
@@ -1,3 +1,187 @@
-import WebStorage from './WebStorage';
+import * as Logger from '../Logger';
 
-export default WebStorage;
+import PlatformStorage from './platforms';
+import InstanceSync from './InstanceSync';
+import MemoryOnlyProvider from './providers/MemoryOnlyProvider';
+import type StorageProvider from './providers/types';
+
+let provider = PlatformStorage;
+let shouldKeepInstancesSync = false;
+let finishInitalization: (value?: unknown) => void;
+const initPromise = new Promise((resolve) => {
+    finishInitalization = resolve;
+});
+
+type Storage = {
+    getStorageProvider: () => StorageProvider;
+} & Omit<StorageProvider, 'name'>;
+
+/**
+ * Degrade performance by removing the storage provider and only using cache
+ */
+function degradePerformance(error: Error) {
+    Logger.logAlert(`Error while using ${provider.name}. Falling back to only using cache and dropping storage.`);
```

Comment:
NAB, it would be useful to log the error message as well. Without knowing anything about _why_ we fell back to using the cache this alert is not really actionable.

Reply:
I always got tripped up on this too, but it is logged on the very next line 😁

Reply:
Oh, but maybe you mean to send the error as part of `logAlert()`?

Reply:
Yep! That's what I meant. I do see the `console.error()` which is local (good) but might be desirable to have server logs. 

I backpedaled here because I'm not sure if we are going to be tracking the frequency of these errors or doing anything about them.

That said, I also wonder if logging alert will lead to issues that nobody can action... I am not sure if the JS logs create issues or not. Maybe they should and if so - this could probably be a `hmmm()`.

Reply:
@tgolen @marcaaron should I do these changes in follow up PR? 👀

Reply:
Yeah, I think it would probably be good to change this to a `hmmm()` and also log the specific error. It would sure be nice to look at server logs to determine _IF_ the fallback is happening and also _WHY_.

Reply:
@tgolen make sense! I've created a follow up PR: https://github.com/Expensify/react-native-onyx/pull/524